lightning/src/ln/channel.rs

```diff
@@ -3771,6 +3969,56 @@ impl<SP: Deref> ChannelContext<SP> where SP::Target: SignerProvider {
 		self.channel_transaction_parameters.channel_type_features = self.channel_type.clone();
 		Ok(())
 	}
+
+	/// Asserts that the commitment tx numbers have not advanced from their initial number.
+	fn assert_no_commitment_advancement(&self) {
+		if self.commitment_secrets.get_min_seen_secret() != (1 << 48) ||
+				self.cur_counterparty_commitment_transaction_number != INITIAL_COMMITMENT_NUMBER ||
+				self.holder_commitment_point.transaction_number() != INITIAL_COMMITMENT_NUMBER {
+			debug_assert!(false, "Should not have advanced channel commitment tx numbers prior to funding_created");
+		}
+	}
+
+	fn get_initial_counterparty_commitment_signature<L: Deref>(
+		&self, logger: &L
+	) -> Result<Signature, ChannelError>
+	where
+		SP::Target: SignerProvider,
+		L::Target: Logger
+	{
+		let counterparty_keys = self.build_remote_transaction_keys();
+		let counterparty_initial_commitment_tx = self.build_commitment_transaction(
+			self.cur_counterparty_commitment_transaction_number, &counterparty_keys, false, false, logger).tx;
+		match self.holder_signer {
+			// TODO (taproot|arik): move match into calling method for Taproot
+			ChannelSignerType::Ecdsa(ref ecdsa) => {
+				ecdsa.sign_counterparty_commitment(&counterparty_initial_commitment_tx, Vec::new(), Vec::new(), &self.secp_ctx)
+					.map(|(signature, _)| signature)
+					.map_err(|_| ChannelError::Close(
```

Comment:
I think it's best to get parity in this PR. I don't believe it will be that much extra work and we can DRY up as you said.

https://github.com/lightningdevkit/rust-lightning/pull/3137/files#r1831690110

Reply:
Alright I'll leave this until I've fixed everything else here and then decide.

Reply:
https://github.com/lightningdevkit/rust-lightning/issues/3404